File: python/circus.py
from ianimal import IAnimal
from arena import Arena

class Circus:
    def __init__(self):
        self.animals: list[IAnimal] = []

    def perform(self, arena: Arena):
        for animal in self.animals:
            print(arena.announce_animal(animal.get_name(), animal.get_species_name()))
            print(arena.display_performance(animal.make_sound()))

File: java/Circus.java
import java.util.ArrayList;
import java.util.List;

public class Circus {
    public List<IAnimal> animals = new ArrayList<>();

    public void perform(Arena arena) {
        for (IAnimal animal : animals) {
            System.out.println(arena.announceAnimal(animal.getName(), animal.getSpeciesName()));
            System.out.println(arena.displayAnimalPerformance(animal.makeSound()));
        }
    }
}
